docs/server/upgrading.rst

```diff
@@ -3,170 +3,222 @@
+upgrading from.
 
-* Familiarize yourself with :doc:`important changes </releases/index>` in
-  Pootle.
 
-* If you want to change the database backend then have a look at the
-  :doc:`database migration <database_migration>` page first. We discourage
-  using SQLite, so if you are using it please migrate to a real database
-  server.
+.. _upgrading#requirements:
 
-* Ensure that you meet the :ref:`hardware requirements
-  <installation#hardware_requirements>` for the newer version.
+System requirements
+-------------------
 
-* Always make backups of all your translation files (your whole
-  :setting:`POOTLE_TRANSLATION_DIRECTORY`). Use the :djadmin:`sync_stores`
-  command to synchronize all your translation files to disk before making any
-  backup.
+You should check that you have all of the necessary
+:ref:`system requirements <requirements>` and installed all required
+:ref:`system packages <requirements#packages>`.
 
-* Also backup your settings, to avoid losing any settings customizations.
+.. warning::
 
-* Make a backup of your complete database using the appropriate *dump*
-  command for your database system. For example :command:`mysqldump` for MySQL,
-  or :command:`pg_dump` for PostgreSQL.
+   Pootle 2.7.0 or newer requires **Python 2.7**
 
-* And don't forget to backup any code, templates or styling customization that
-  you have done to your installation.
+   If you are upgrading from a virtual environment using an earlier python
+   version, you **must** upgrade or rebuild your virtual environment first.
 
-* Familiarize yourself with any new :ref:`settings <settings#available>` that
-  have been introduced.
 
+.. _upgrading#clean-bytecode:
 
-.. _upgrading#upgrading:
+Clean up stale python bytecode
+------------------------------
 
-Upgrading
----------
+You should remove any stale python bytecode files before upgrading.
+
+Assuming you are in the root of your virtualenv folder you can run:
+
+.. code-block:: bash
+
+   (env) $ find . -name ".pyc" | xargs rm
 
-Upgrading Pootle using the :command:`pip`.
 
-.. note:: You will need to adjust these instructions if you installed Pootle
-   using another method, such as directly from a Git checkout.
+.. _upgrading#upgrading-2.6:
+
+
+Upgrading from a version prior to 2.6
+-------------------------------------
+
+If you are upgrading from a version prior to 2.6 you will need to first upgrade
+to the latest 2.6.x version and then you will be able to upgrade to the latest
+version.
+
+.. code-block:: bash
+
+   (env) $ pip install --upgrade "Pootle>=2.6,<2.7"
+   (env) $ pootle setup
 
 .. warning::
+   The 2.6.x releases are meant only as a migration step.
+
+   You should upgrade immediately to the latest version once setup has
+   completed.
+
+
+.. _upgrading#upgrading-latest:
+
+Upgrading from version 2.6.x
+----------------------------
+
+Upgrade to the latest Pootle package:
+
+.. code-block:: bash
+
+   (env) $ pip install --upgrade Pootle
```

Comment:
I think we want to do a ``pyclean`` before this no? If I recall stale ``*.pyc`` in migration code has caused us some pain before.